pkgs/by-name/sy/sysdig-cli-scanner/package.nix

```diff
@@ -0,0 +1,55 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  makeWrapper,
+}:
+let
+  versionMetadata = import ./sysdig-cli-scanner.versions.nix;
+  fetchForSystem = versionMetadata.${stdenv.system} or (throw "unsupported system ${stdenv.system}");
+in
+stdenv.mkDerivation {
+  pname = "sysdig-cli-scanner";
+  version = versionMetadata.version;
+
+  src = fetchurl { inherit (fetchForSystem) url hash; };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  dontUnpack = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D $src $out/bin/sysdig-cli-scanner
+
+    wrapProgram $out/bin/sysdig-cli-scanner \
+      --add-flags --dbpath="/tmp/sysdig-cli-scanner/"
```

Comment:
Do we really want to use a predictable directory for that?
Also people using a tmpfs under /tmp won't benefit from this.

Reply:
If `dbpath` is not specified, it tries to use the same path of the binary, which is the path under `/nix/store`, which is not writable, and therefore it fails to run.

Users using a tmpfs under `/tmp` should not have any problem, the DB is updated every 12 hours so it has to be replaced anyways after some time. I can modify it to use `~/.cache/sysdig-cli-scanner` instead if you think that's better.

Reply:
+1 on using `~/.cache`

Reply:
Unfortunately, after some tests I did, if you specify 

```
    wrapProgram $out/bin/sysdig-cli-scanner \
      --add-flags --dbpath="~/.cache/sysdig-cli-scanner/"
```

The result is that a `~/.cache` is created in the current execution path, not in the home.

<details>
<summary>
See more details here:
</summary>
<pre><code>❯ nixpkgs-review rev HEAD

$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
remote: Enumerating objects: 4224, done.
remote: Counting objects: 100% (2441/2441), done.
remote: Compressing objects: 100% (183/183), done.
remote: Total 4224 (delta 2308), reused 2326 (delta 2242), pack-reused 1783
Receiving objects: 100% (4224/4224), 2.87 MiB | 3.23 MiB/s, done.
Resolving deltas: 100% (2904/2904), completed with 743 local objects.
From https://github.com/NixOS/nixpkgs
   de7067cf2f59..e9c63d27ca2d  master     -> refs/nixpkgs-review/0
$ git worktree add /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs e9c63d27ca2d4ec12b96d21393a00aa3cdda40aa
Preparing worktree (detached HEAD e9c63d27ca2d)
Updating files: 100% (42758/42758), done.
HEAD is now at e9c63d27ca2d Merge pull request #323988 from NyCodeGHG/gauge-patchelf
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs nixpkgs-overlays=/tmp/tmpenkhg6v8 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 9be1a0dcd204b16534f5aaa93d188322afb1979d
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs nixpkgs-overlays=/tmp/tmpenkhg6v8 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
1 package added:
sysdig-cli-scanner (init at 1.13.2)

$ nix build --file /nix/store/ww9sz3m14wsh484vchhgws9vhg2mlfcx-nixpkgs-review-2.10.4/lib/python3.11/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path nixpkgs=/home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs nixpkgs-overlays=/tmp/tmpenkhg6v8 --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr system x86_64-linux --argstr nixpkgs-path /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs --argstr nixpkgs-config-path /tmp/tmpl6dqpb7m.nix --argstr attrs-path /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/attrs.nix
1 package built:
sysdig-cli-scanner

$ /nix/store/wxp968584yl2ygfm3kwhl712wc2y6r5k-nix-2.18.5/bin/nix-shell --argstr system x86_64-linux --argstr nixpkgs-path /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs --argstr nixpkgs-config-path /tmp/tmpl6dqpb7m.nix --argstr attrs-path /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/attrs.nix --nix-path nixpkgs=/home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/nixpkgs nixpkgs-overlays=/tmp/tmpenkhg6v8 /nix/store/ww9sz3m14wsh484vchhgws9vhg2mlfcx-nixpkgs-review-2.10.4/lib/python3.11/site-packages/nixpkgs_review/nix/review-shell.nix

❯ sysdig-cli-scanner -a https://us2.app.sysdig.com/ quay.io/sysdig/cloud-connector
2024-08-09T09:38:29+02:00 Starting analysis with Sysdig scanner version 1.13.2
2024-08-09T09:38:30+02:00 Retrieving MainDB...
2024-08-09T09:42:38+02:00 Done 167.8 MB
2024-08-09T09:42:39+02:00 Loading MainDB...
2024-08-09T09:42:39+02:00 Done
2024-08-09T09:42:40+02:00 Retrieving image...
INFO[0250] overlay: test mount did not indicate whether or not metacopy is being used: failed to mount overlay for metacopy check with "" options: operation not permitted 
2024-08-09T09:42:52+02:00 Done
2024-08-09T09:42:52+02:00 Scan started...
2024-08-09T09:42:54+02:00 Uploading image sbom to backend...
2024-08-09T09:42:54+02:00 Uploading scanresult with sbom to backend...
2024-08-09T09:42:56+02:00 Done
2024-08-09T09:42:56+02:00 Total execution time 4m26.999835227s

Type: dockerImage
ImageID: sha256:7f8b0cc914659812ddab3994880b5b9b75c20a0ea4b1fbebbd94f79c3091c34b
Digest: sha256:b39c308f67c4f6984fc0e2423f5dc4ae1d7c9131d1074bf6e63d563a4d37fdae
BaseOS: redhat 8.10
PullString: quay.io/sysdig/cloud-connector

30 vulnerabilities found
0 Critical (0 fixable)
0 High (0 fixable)
1 Medium (0 fixable)
26 Low (0 fixable)
3 Negligible (0 fixable)

                                    POLICIES EVALUATION
    Policy: Critical Vulnerability Found          PASSED (0 failures - 0 risks accepted)
    Policy: Sensitive Information or Secret Found PASSED (0 failures - 0 risks accepted)

Policies evaluation PASSED at 2024-08-09T09:42:56+02:00
Full image results here: https://us2.app.sysdig.com/secure/#/vulnerabilities/results/17e9ff9b48d8eb6fe886aff929a97a30/overview (id 17e9ff9b48d8eb6fe886aff929a97a30)
Execution logs written to: /home/sysdig/.cache/nixpkgs-review/rev-9be1a0dcd204b16534f5aaa93d188322afb1979d/scan-logs

❯ ls -al
total 20
drwxr-xr-x 1 sysdig users  114 ago  9 09:38  .
drwxr-xr-x 1 sysdig users  268 ago  9 09:32  ..
drwxr-xr-x 1 sysdig users   12 ago  9 09:38 '~'
-rw-r--r-- 1 sysdig users   41 ago  9 09:37  attrs.nix
drwxr-xr-x 1 sysdig users   44 ago  9 09:37  logs
drwxr-xr-x 1 sysdig users  348 ago  9 09:32  nixpkgs
-rw-r--r-- 1 sysdig users  233 ago  9 09:37  report.json
-rw-r--r-- 1 sysdig users  198 ago  9 09:37  report.md
drwxr-xr-x 1 sysdig users   36 ago  9 09:37  results
-rw-r--r-- 1 sysdig users 4900 ago  9 09:42  scan-logs</code></pre>
</details>

See the `drwxr-xr-x 1 sysdig users   12 ago  9 09:38 '~'` line. Unfortunately, I think we will need to leave it to `/tmp`, unless you know how to reference the actual home.

Reply:
How about using `$HOME` (might need to be escaped `\$HOME`) instead of `~`? Does that work, @tembleking?

Reply:
That one works! I am commiting it now, thanks for the suggestion!

Reply:
What do you think about the change folks? @afh @SuperSandro2000

Reply:
Using ~/.cache for this feels a lot nicer